### PR TITLE
 #425 Routing implementation

### PR DESCRIPTION
**Pull request by stephanevieira75**
ref to  #425

### CODE REVIEW

File: twake/frontend/src/app/app.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  }, []);
+
+  return (
+    <Integration>
+      <Router history={RouterServices.history}>
+        <Switch>
+          {RouterServices.routes.map((route: any, index: number) => {
+            return (
+              <Route
+                key={route.name + ' ' + index}
```
Comment (chamerling) [inline twake/frontend/src/app/app.tsx:22]: Avoir string concatenation when possible, use string literals, it makes clean code

File: twake/frontend/src/app/scenes/Error/Error.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    <div className="full_page_error" key="page_error">
+      <div className="error_message skew_in_top_nobounce">
+        <div className="title">
+          <Emojione type=":boom:" /> {Languages.t('scenes.aie', [], 'Aïe !')}
+        </div>
+        <div className="subtitle">
+          {Languages.t(
+            'scenes.error_on_twake',
+            [],
+            'Vous avez trouvé une erreur sur la plateforme Twake !',
```
Comment (chamerling) [inline twake/frontend/src/app/scenes/Error/Error.tsx:28]: Why not having them in EN by default?

File: twake/frontend/src/app/scenes/Error/ErrorBoundary.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import RouterServices from '../../services/RouterServices';
+import 'app/ui.scss';
+
+type State = {
+  hasError: boolean;
+  error: any;
+};
+
+export default class ErrorBoundary extends React.Component<{}, State> {
+  constructor(props: any) {
```
Comment (chamerling) [inline twake/frontend/src/app/scenes/Error/ErrorBoundary.tsx:11]: I am not a React dev, but can't we have a real type here?

File: twake/frontend/src/app/scenes/Setup/Setup.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+type ReadyState =
+  | {
+      elasticsearch_connection: boolean;
+      elasticsearch_mapping: boolean;
+      db_connection: boolean;
+      db_mapping: boolean;
+      init: boolean;
+    }
+  | true;
```
Comment (chamerling) [inline twake/frontend/src/app/scenes/Setup/Setup.tsx:16]: Does it means that it is true when every state above is true? You should not define types like this which are changing based on condition. Instead you have to have a `isReady()` function which tells you is it is ready based on the ReadyState.

File: twake/frontend/src/app/services/Globals.js [hunk 5]
```diff
@@ -203,8 +224,8 @@ class Globals {
   }
 
   localStorageGetItem(key, callback) {
-      // eslint-disable-next-line no-undef
-      if (typeof SyncStorage != 'undefined') {
+    // eslint-disable-next-line no-undef
+    if (typeof SyncStorage != 'undefined') {
```
Comment (chamerling) [inline twake/frontend/src/app/services/Globals.js:228]: `if (SyncStorage) {` is generally enough

File: twake/frontend/src/app/services/RouterServices.tsx [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+  routes?: Route[];
+  component?: any;
+  options?: any;
+};
+
+class RouterServices {
+  history = createBrowserHistory();
+
+  // Define your route here
+  pathnames: Readonly<any> = {
```
Comment (chamerling) [inline twake/frontend/src/app/services/RouterServices.tsx:23]: `Readonly<string>`

File: twake/frontend/src/app/scenes/Error/Error.tsx (hunk 2 above)
Comment (RomaricMourgues) [reply twake/frontend/src/app/scenes/Error/Error.tsx:28]: Old component from startup, when all of our clients were french.
You can translate it or remove the fallback Stephane.

File: twake/frontend/src/app/scenes/Error/ErrorBoundary.tsx (hunk 3 above)
Comment (RomaricMourgues) [reply twake/frontend/src/app/scenes/Error/ErrorBoundary.tsx:11]: This state can be removed. We don't use this state.

Comment (RomaricMourgues) [reply twake/frontend/src/app/scenes/Error/ErrorBoundary.tsx:11]: (for the error: any state)
For the (props: any), the type is "{}" like in Component<{}, State> @stephanevieira75 
So (props: {})

File: twake/frontend/src/app/services/Globals.js (hunk 5 above)
Comment (RomaricMourgues) [reply twake/frontend/src/app/services/Globals.js:228]: Not in the scope of your issue @stephanevieira75

Comment (RomaricMourgues) [reply twake/frontend/src/app/services/Globals.js:228]: But thanks for the comment Christophe 👍

**Pull request merged**